docs/Dialects/Interop/RationaleInterop.md

```diff
@@ -0,0 +1,590 @@
+# Interoperability Dialect Rationale
+
+This document describes the various design points of the Interop dialect, a
+dialect that is used to represent partially lowered interoperability layers and
+that provides common interfaces and utilities for automated interop generation.
+This follows in the spirit of other
+[MLIR Rationale docs](https://mlir.llvm.org/docs/Rationale/).
+
+- [Interoperability Dialect Rationale](#interoperability-dialect-rationale)
+  - [Introduction](#introduction)
+  - [Procedural Interop](#procedural-interop)
+    - [Representing Partial Lowerings](#representing-partial-lowerings)
+    - [Interop Mechanisms](#interop-mechanisms)
+    - [Instance-side Lowering](#instance-side-lowering)
+    - [Container-side Lowering](#container-side-lowering)
+    - [Bridging between Interop Mechanisms](#bridging-between-interop-mechanism)
+    - [How to use this dialect](#how-to-use-this-dialect)
+    - [Design considerations](#design-considerations)
+
+## Introduction
+
+The increasing number of CIRCT-based backends as well as the usage of many
+external tools raises questions about interoperability and composition of
+those tools. Are we supposed to write interop-layers between them by hand or
+can this also be automatically generated? Can we generate interfaces to the
+outside in multiple standardized formats/languages automatically?
+
+Let's pick simulation as a concrete example. We have Verilator, CIRCT-based
+simulators like the event-driven `llhd-sim`, and at some point maybe another
+non-event-driven simulator to compete with Verilator in terms of performance.
```

Comment:
The PR can just mention "an event-driven simulator inside CIRCT" without calling out Verilator competition specifically.